### PR TITLE
Feature-add-basic-UI

### PR DESCRIPTION
**Pull request by maximoortelli**
### Description:
--------------------------------------------------------------------------------------------------------------------------------------------

- This pull request introduces a new feature, Basic-UI, to enhance the school library app by providing a user-friendly form of interaction. The primary goal of this feature is to allow the app to be invoked as an executable, enhancing its usability beyond IRB.

### Changes Made:
--------------------------------------------------------------------------------------------------------------------------------------------

-  Create a user interface for the school library app, enabling users to interact with the app in a more intuitive manner.

-  Implement the UI based on the example provided in the video, ensuring consistency and familiarity for users.

General Requirements:
--------------------------------------------------------------------------------------------------------------------------------------------

-  Ensure the code adheres to linting standards, with no linter errors or warnings present.

-  Follow Gitflow methodology, maintaining separate branches for feature development, and using proper branching conventions.

-  Document the implementation process professionally, providing clear and concise comments within the codebase.

-  Adhere to the list of best practices for Ruby programming, maintaining clean and readable code.

Project Requirements:
--------------------------------------------------------------------------------------------------------------------------------------------

-  Watch the UI example video to understand the desired behavior and user experience.
- Create an app.rb file to serve as the entry point for the console app, containing methods as per the specified requirements.
- Implement methods to list books, list people, create a person (teacher or student), create a book, create a rental, and list all rentals for a given person id.
- Define the main method in main.rb as the entry point for user interaction.
- Present users with a menu of options, allowing them to choose actions to perform.
- Prompt users for any necessary parameters required for each action.
- Provide a way for users to quit the app gracefully._

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+require_relative 'classroom'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
```
Comment (TadesseDev) [inline app.rb:15]: - [x] Dear @maximoortelli, Good job adding the `app.rb` and `main.rb` files, but currently I could not get your app to run, I see the first and obvious issue with your `rental.rb` file having the error `undefined local variable or method 'person' for Rental:Class (NameError)` I address this but the app keeps throwing error one after the other. Please make sure the app is working fine and re-submit for a re-review. 

### The issue 

https://github.com/maximoortelli/OOP_School_Library/assets/69077061/4bfcd035-216c-4bcd-9555-5a4f1b54cca5

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @maximoortelli 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Good job adding the `app` and `main` files. ✅ 
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (maximoortelli) [reply app.rb:15]: Done, thank you for the review. Im fixed the bugs.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @maximoortelli 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x]  _Check the inline comments under the review for sections to improve._
- [x] NB: I am adding this recommendation here because `Teacher.rb` file is not added to the file changes!
- [x] [REQUIRED] You did an excellent job resolving the bug and bringing the app back to live, however there is one thing left. I am unable to state/add a Teacher's specification. 

### Here is the error:
Screenshot 2023-11-09 at 17 14 09

### To resolve this
 In the `create_teacher` method, you are creating a new Teacher object with the provided name, age, and specialization. Let's ensure that the Teacher class constructor can handle these arguments correctly. 
 
 ``` ruby
 def initialize(name, age, specialization, permission: true)
    super(name, age, parent_permission: permission)
    @specialization = specialization
  end
```

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @maximoortelli 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**